src/elements/content-sharing/SharingNotification.js

```diff
@@ -239,6 +248,43 @@ function SharingNotification({
         setCollaboratorsList,
     ]);
 
+    // Set the getContacts function, which is used for inviting collaborators in the USM
+    React.useEffect(() => {
```

Comment:
tbh, having these large `useEffect` calls make it a bit hard to read the `SharingNotification` component. Would it make sense to create custom hooks that could be reused by other components and simplify this component? Maybe a `useContacts` and `useCollaborators` hooks?

Reply:
Good idea. Will look into refactoring this component.